src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+$ export COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT=40s
+$ cockroach start --certs-dir certs --listen-addr=:26257 --http-addr=:26258 --advertise-addr=10.150.0.69:26257 --join=34.48.91.20:26257 --store path=/mnt/data1/cockroach,attrs=store1:node1:node1store1 --store path=/mnt/data2/cockroach,attrs=store2:node1:node1store2 --store path=/mnt/data3/cockroach,attrs=store3:node1:node1store3 --store path=/mnt/data4/cockroach,attrs=store4:node1:node1store4 --locality=cloud=gce,region=us-east4,zone=us-east4-a,rack=0 --wal-failover=among-stores --log-config-file=logs.yaml
+~~~
+
+Notice the flags passed to `cockroach start`:
+
+- Multiple `--store` paths
+- `--wal-failover=among-stores` since we have multiple stores to choose from for WAL failover
+- `--log-config-file` uses the `logs.yaml` we created in [Step 3](#3-log-configuration-for-wal-failover)
+
+## WAL Failover in action
+
+The instructions in this section show how to trigger WAL failover by introducing disk stalls on one of the nodes in the cluster. Note that the steps are the same for [single-store](#single-store-config) or [multi-store](#multi-store-config) clusters.
+
+### 1. Set up a node with scripts to trigger WAL Failover
+
+We will trigger WAL Failover on a node by slowing down the read bytes per second (`rbps`) and write bytes per second (`wbps`) for a disk.
+
+`ssh` into the node whose disk you want to stall and unstall.
+
+Run the `lsblk` command, and look for `MAJ:MIN` values for `/mnt/data1` in `lsblk` output. In this case it is `259:0`; we will use this for the disk stall/unstall scripts.
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+roachprod ssh $CLUSTER:1 lsblk
+NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
+loop0     7:0    0  63.9M  1 loop /snap/core20/2182
+loop1     7:1    0 354.3M  1 loop /snap/google-cloud-cli/223
+loop2     7:2    0    87M  1 loop /snap/lxd/27428
+loop3     7:3    0  39.1M  1 loop /snap/snapd/21184
+sda       8:0    0    10G  0 disk
+├─sda1    8:1    0   9.9G  0 part /
+├─sda14   8:14   0     4M  0 part
+└─sda15   8:15   0   106M  0 part /boot/efi
+nvme0n1 259:0    0   375G  0 disk /mnt/data1
+nvme0n2 259:1    0   375G  0 disk /mnt/data2
+nvme0n3 259:2    0   375G  0 disk /mnt/data3
+nvme0n4 259:3    0   375G  0 disk /mnt/data4
+~~~
+
+Next, we will create scripts for stalling and unstalling the disk on this node. The script will add a line in `/sys/fs/cgroup/system.slice/cockroach-system.service/io.max` to change the [cgroup](https://en.wikipedia.org/wiki/Cgroups) settings "read bytes per second" (`rbps`) and "write bytes per second" (`wbps`) of the disk to a very low value for stalling, and a max value for unstalling.
+
+Create text files with the commands that change the cgroup settings:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+echo '259:0 rbps=4 wbps=4' > stall.txt
+echo '259:0 rbps=max wbps=max' > unstall.txt
+echo 'wq! /sys/fs/cgroup/system.slice/cockroach-system.service/io.max' > cmd.txt
+~~~
```

Comment:
I don't think we should use these scripts. It's very strange to use vim in this way.

The user should be able to run
```
sudo echo '259:0 rbps=4 wbps=4' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
```
to "stall" the device and
```
sudo echo '259:0 rbps=max wbps=max' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
```
to unstall.

Reply:
i've removed `stall.txt` `unstall.txt`, and `cmd.txt`

and replaced the use of passing `stall.txt` to `vim` to use the `echo` statements you provided for stalling and unstalling instead